core/pkg/artifacts/saver.go

```diff
@@ -123,17 +138,11 @@ func (as *ArtifactSaver) createManifest(
 	return response.GetCreateArtifactManifest().ArtifactManifest, nil
 }
 
-func (as *ArtifactSaver) uploadFiles(artifactID string, manifest *Manifest, manifestID string, _ chan<- *service.Record) error {
-	const batchSize int = 10000
-	const maxBacklog int = 10000
-
-	type TaskResult struct {
-		Task *filetransfer.Task
-		Name string
-	}
-
-	// Prepare all file specs.
-	var fileSpecs []gql.CreateArtifactFileSpecInput
+func (as *ArtifactSaver) uploadFiles(
+	artifactID string, manifest *Manifest, manifestID string, _ chan<- *service.Record,
+) error {
+	// Prepare GQL input for files that (might) need to be uploaded.
+	fileSpecs := map[string]gql.CreateArtifactFileSpecInput{}
```

Comment:
minor nit: change these to `namedFileSpecs` to match convention in `processFiles`